### PR TITLE
Make use of DownloadLink inside of HTMLSerializer

### PR DESCRIPTION
**Pull request by AMBacelar**
## Who is this for?
this PR addresses [this conversation](https://github.com/wellcomecollection/wellcomecollection.org/issues/8444) 

## What is it doing for them?
- DownloadLink component is moved to common
- HTMLSerializer makes use of the now moved DownloadLink component when the type of link is a document
- added a utility function to get the appropriate mime type from the file extension.
- refactored DownloadLink component to make use of `children` prop _OR_ `labelText` prop
- updated the imports for DownloadLink

### CODE REVIEW

File: cardigan/stories/components/DownloadLink/DownloadLink.stories.tsx [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import DownloadLink from '../../../../catalogue/webapp/components/DownloadLink/DownloadLink';
+import DownloadLink from '../../../../common/views/components/DownloadLink/DownloadLink';
```
Comment (rcantin-w) [inline cardigan/stories/components/DownloadLink/DownloadLink.stories.tsx:1]: I know it wasn't used here before but we have an alias/shortcut for these if you want! `@weco/common/views/components/DownloadLink/DownloadLink`

File: catalogue/webapp/components/AudioList/AudioList.tsx [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import { FC } from 'react';
 import Space from '@weco/common/views/components/styled/Space';
 import AudioPlayer from '@weco/common/views/components/AudioPlayer/AudioPlayer';
-import DownloadLink from '../DownloadLink/DownloadLink';
+import DownloadLink from '@weco/common/views/components/DownloadLink/DownloadLink';
```
Comment (rcantin-w) [inline catalogue/webapp/components/AudioList/AudioList.tsx:4]: Ah great 😁

File: common/views/components/DownloadLink/DownloadLink.tsx [hunk 3]
```diff
@@ -52,13 +72,21 @@ const DownloadLink: FunctionComponent<Props> = ({
...
     <span className="flex-inline flex--v-center">
       <Icon icon={download} />
-      <span className="underline-on-hover">{linkText}</span>
-      {format && <Format as="span">{format}</Format>}
+      <TextToDisplay>{linkText || children}</TextToDisplay>
+      {format && (
+        <Space
+          as="span"
+          h={{ size: 'm', properties: ['margin-left'] }}
+          className={`${font('intb', 5)} font-neutral-600`}
```
Comment (rcantin-w) [inline common/views/components/DownloadLink/DownloadLink.tsx:85]: I've merged in a change yesterday where we don't declaire `font-[color]` as a styling class anymore, so better to style it differently using `props.theme.color('neutral.600')` somehow.

File: content/webapp/components/HTMLSerializers/HTMLSerializers.tsx [hunk 4]
```diff
@@ -121,64 +119,21 @@ export const defaultSerializer: JSXFunctionSerializer = (
...
-          >
-            <span
-              style={{
-                top: '8px',
-                display: 'inline-block',
-                height: '24px',
-                width: '24px',
-                position: 'relative',
-                userSelect: 'none',
+          <>
```
Comment (rcantin-w) [inline content/webapp/components/HTMLSerializers/HTMLSerializers.tsx:122]: Is the Fragment needed?

File: cardigan/stories/components/DownloadLink/DownloadLink.stories.tsx (hunk 1 above)
Comment (AMBacelar) [reply cardigan/stories/components/DownloadLink/DownloadLink.stories.tsx:1]: does this apply to the `.mdx` file too?

File: content/webapp/components/HTMLSerializers/HTMLSerializers.tsx (hunk 4 above)
Comment (AMBacelar) [reply content/webapp/components/HTMLSerializers/HTMLSerializers.tsx:122]: nope, I'll remove it

File: cardigan/stories/components/DownloadLink/DownloadLink.stories.tsx (hunk 1 above)
Comment (rcantin-w) [reply cardigan/stories/components/DownloadLink/DownloadLink.stories.tsx:1]: Yeah, should work there too! It's honestly nitpicking, so up to you 👍

Comment (AMBacelar) [reply cardigan/stories/components/DownloadLink/DownloadLink.stories.tsx]: changes made!

**Comment by rcantin-w** [comment]
Thank you, we're definitely getting a cleaner use of this!

The Chromatic build is failing (see the builds below), you can track the error by clicking on "Details" next to it and see what's happening. As always, am available if there's anything!

If we look at the example in the ticket (`http://localhost:3000/pages/YFiLgBAAACoAKF7b`), the link would ideally be aligned with the rest of the copy and has lost some information:

[Prod](https://wellcomecollection.org/pages/YFiLgBAAACoAKF7b)
Screenshot 2022-10-14 at 12 47 31

Branch
Screenshot 2022-10-14 at 12 47 45

Apologies, we should've discussed what we wanted the final result to look/be formatted like in both cases! I think we keep the parentheses around the format/weight, bold seems alright to me as long as it's in line with the rest of the copy/font-size. What do you think?

**Comment by AMBacelar** [comment]
Screenshot 2022-10-17 at 11 12 59
I went for having the LinkText as unstyled as possible when used in lieu of the HTML serialiser, so that it looks as close to what it was as possible

**Review by rcantin-w — APPROVED** [review]
Apart from our convo on the new trackingEvent, I think let's go for it! 🙌

**Pull request merged**